### ISSUE TITLE
[Feature]: Support further emulations in Podman Machine

### ISSUE THREAD

**Issue by FlorianLoch** [containers/podman#17267]
### Feature request description

Podman Machine already supports and allows running/building containers for different platforms using QEMU.

If I understand correctly, this works by using `qemu-user-static` and registering support for additional binary formats via `binfmt`.

This basically works fine and allows me, e.g., to build and run `amd64` containers on an `aarch64` M1 Mac.

I am asking for adding support of additional QEMU emulators, especially support for still quite common platforms like `arm32v7` as used by Raspberry PI 3. 
Docker for Mac includes these in their VM image, therefore building such images for me is currently only possible via Docker for Mac - but I would love to move away from it and be able to use Podman only.

### Suggest potential solution

As Docker for Mac supports this, I am quite sure it could simply be solved by installing the additional packages into Podman Machine's Fedora image and registering the binary formats. 

Also, I tried this locally via `podman machine ssh` and was not able to make it work straight away.
But that was probably just some mistake from my side.

I agree, that this might increase the size of the VM's image and might therefore be disliked by some users - in that case I think it would be great to provide some script/documentation on how to augment the VM's image in order to support additional emulations.

**Comment by rhatdan** [comment]
This is more under the control of Fedora CoreOS (Used in Podman machine for MAC).  We can open a request to add the package.  Eventually we need to just layer packages on Fedora Coreos needed for podman machine rather then always having to go back to CoreOS for the updates

**Comment by rhatdan** [comment]
@dustymabe WDYT?

**Comment by FlorianLoch** [comment]
Ah, I thought Podman Machine pulls a custom image for the VM, which could be extended to contain these package layers. 

But yes, if it comes directly from Fedora that wouldn't work that easily :/

**Comment by dustymabe** [comment]
I doubt it would be something we'd add to the base FCOS but you can always make the request.

For now you can just package layer it with something like `sudo rpm-ostree install XYZ && sudo reboot` where `XYZ` is the package name. I imagine the package you are looking for is in [this list](https://koji.fedoraproject.org/koji/search?terms=qemu-7.0.0-13.fc37&type=build&match=exact).

**Comment by FlorianLoch** [comment]
I took another look at that list and ran:

```shell
sudo rpm-ostree install qemu-user-static-arm qemu-system-arm qemu-user-binfmt
sudo systemctl restart systemd-binfmt.service
sudo systemctl reboot
```

But that didn't get me any further, still cannot run my `arm32v7` image via `podman build --platform linux/arm/v7 .`. 
With `Docker for macOS` it is working - and AFAIK they are also using QEMU inside a VM for this...

So I must be missing something...

This is probably out of scope for this issue, but any idea what could be the problem here?

I agree, theses layers do not need to be added to the base image, but documenting how to properly set up this additional arch support would be really nice as it really isn't obvious to me (and probably others).

**Comment by dustymabe** [comment]
> This is probably out of scope for this issue, but any idea what could be the problem here?

hmm. I would have thought that would work. Maybe someone else here knows more or you can file an issue against [`qemu`](https://src.fedoraproject.org/rpms/qemu) in Fedora (click on the `Bug Reports` link).

**Comment by baude** [comment]
hello @FlorianLoch thanks for the RFE.  We are in fact working on a solution for the things you are asking about.  You should see it hit upstream very soon.  No need to alter the official fcos for this.

**Comment by baude** [comment]
this also should be of interest @FlorianLoch

**Comment by FlorianLoch** [comment]
Hi @baude 
sounds good, looking forward to this. 

But as said, my intention was never to change/request a change to the official FCOS image - I just wasn't aware, that Podman Machine fetches this one. I thought that would already be a custom fit, patched one.

Did you possibly miss to include a link in your second comment? ;)

**Comment by rhatdan** [comment]
@baude this is waiting for a response from you I believe.

**Comment by Luap99** [comment]
Or own images (quay.io/podman/machine-os) contain the full qemu-user-static package so this was fixed with 5.0 I would say
